### PR TITLE
rules: Out of the box ruleset for OKTA Falco Plugin

### PR DESCRIPTION
**Pull request by darryk10**
Creating out of the box rules for the new OKTA Faclo plugins
Opened PR: https://github.com/falcosecurity/plugins/pull/81
Signed-off-by: darryk10 <stefano.chierici@sysdig.com>

**What type of PR is this?**
This PR introduce the out of the box ruleset for the new OKTA Falco plugin created.
Reference https://github.com/falcosecurity/plugins/pull/81
> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

> If contributing rules or changes to rules, please make sure to also uncomment one of the following line:

> /kind rule-update

/kind rule-create

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area engine

/area rules

> /area tests

> /area proposals

**What this PR does / why we need it**:
This PR add the out of the box ruleset for the new OKTA Falco plugin
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
Depends on opened PR https://github.com/falcosecurity/plugins/pull/81
**Does this PR introduce a user-facing change?**:

```release-note
WIP: Out of the box ruleset for OKTA Falco Plugin
```

### CODE REVIEW

File: rules/okta_rules.yaml
Comment (Issif) [inline rules/okta_rules.yaml:29]:
```suggestion
  output: "A user has accessed an app using OKTA (user=%okta.actor.name, app=%okta.app)"
```

**Review by Issif — CHANGES_REQUESTED** [review]

**Comment by jasondellaluce** [comment]
I see the last two commits have a `Co-authored-by: ...` but not a `Signed-off-by: ...`, which is breaking DCO. My guess is that this happened through a GitHub PR suggestion. I fell with this trick lots of times, please remember of always putting your `Signed-off-by: ...` before committing a suggestion! In this case you need to add it to those commits in order to unblock the DCO. Thank you! 😄

File: rules/okta_rules.yaml
Comment (mstemm) [inline rules/okta_rules.yaml:10]: I think overall the rules do a good job at showing up what info the plugin provides, but we should decide whether the rules are supposed to be opinionated and look for suspicious/alarming events, or rather provide a changelog of information where a user would do additional analysis on the changelog to find suspicious/alarming events.

The distinction is important because I've seen users, for example, just blindly enable all the k8s audit rules "to provide the best coverage", and then they're drowning in falco alerts that denote all changes to their k8s cluster. And then they blame falco for all the "noise".

I think we should make this distinction at least using rule tags and comments.

Comment (darryk10) [reply rules/okta_rules.yaml:10]: thanks for your review.
All the rules due to the types of logs and the type of Okta service (IAM) are basically audit rules. 
The response for each rule might change user by user based on what they think it's important for their environment. Hard to fine a one fit all behaviour.
For instance here is the suggestion from OKTA on what it should be monitored from the security prospective: https://github.com/OktaSecurityLabs/CheatSheets/blob/master/SecurityEvents.md
Some of the one proposes are even more noisy and I haven't included that already (login and log out logs).
I can add an "audit" tag on some of them hard to find which one isn't audit due to the log source.

Comment (darryk10) [reply rules/okta_rules.yaml:10]: @mstemm I disabled by default some rules leaving the most important on security side active. 
The all noisy ones should be disabled by default to reduce the noise as you suggested. 
Thanks for the feedback and let me know your thoughts

**Review by jasondellaluce — APPROVED** [review]
/approve

Thank you @darryk10, I think this is fantastic!

Just leaving a note for us in the future. I think adding rulesets for plugins that are not bundled into Falco is not a scalable pattern. What we would like to have in the future is probably a distribution strategy that would improve the UX and allow users to pull plugins & rulesets from a registry/index. I think this is something we'll want to prioritize in the future! But for sake of simplicity, I'm ok to keep this here!

**Review by mstemm — APPROVED** [review]
Thanks for disabling the audit-type rules. I agree with @jasondellaluce that we'll need to define a process where we decide which rules/plugins are shipped with falco by default, and which are generally available but a user might need to build the plugin themself or add the rules for that plugin's event source by themself.

I'm okay for now with adding the okta rules, because it's a great use case for plugins.

**Pull request merged**
